src/lib/localStorage.ts

```diff
@@ -11,6 +13,34 @@ export function hasLocalStorage(): boolean {
   }
 }
 
-export const localStorage = hasLocalStorage()
-  ? window.localStorage
-  : { getItem: () => null, setItem: () => null, removeItem: () => null }
+export function getLocalStorage(): LocalStorage {
+  return hasLocalStorage()
+    ? window.localStorage
+    : {
+        getItem: () => null,
+        setItem: () => null,
+        removeItem: () => null
+      }
+}
+
+export function migrateStorage<T>(key: string, migrations: Migrations<T>) {
+  let version = 1
+  const localStorage = getLocalStorage()
+  const dataString = localStorage.getItem(key)
+  if (dataString) {
+    const data = JSON.parse(dataString as string)
+
+    if (data.storage) {
+      version = parseInt(data.storage.version || 0, 10) + 1
+    }
+
+    while (migrations[version]) {
+      const newData = migrations[version](data)
+      localStorage.setItem(
+        key,
+        JSON.stringify({ ...(newData as Object), storage: { version } })
```

Comment:
it looks like this line is getting rid of the `loading: true` property that comes with the initial state of the storage reducer (see this line https://github.com/decentraland/decentraland-dapps/pull/11/files#diff-8544c74e5fd9e4b18f35209ce5efd2beR12) I dunno if that breaks anything, probably @NicoSantangelo knows better than myself about this, but it seems like this should rather be:

```ts
JSON.stringify({ ...(newData as Object), storage: { ...newData.storage, version } })
```

Reply:
I think this is not a big deal because the `loading` will typically be set after creating the middleware, as a part of the redux-storage lib.
That said I think it doesn't hurt adding the spread, _just in case_

Also, can we get rid of the as `Object`? Maybe be a bit more specific?